### ISSUE TITLE
wrong regexp in get_backup_file_list

### ISSUE THREAD

**Issue by DmitryFillo** [mjs7231/django-dbbackup#37]
[Regexp contain ^](https://github.com/mjs7231/django-dbbackup/blob/master/dbbackup/management/commands/mediabackup.py#L117), but each file [contain backup directory](https://github.com/mjs7231/django-dbbackup/blob/master/dbbackup/storage/filesystem_storage.py#L44), and so no one old media backup will be deleted.

By the way, [dbbackup.py contains basename function](https://github.com/mjs7231/django-dbbackup/blob/master/dbbackup/management/commands/dbbackup.py#L85).

**Comment by DmitryFillo** [comment]
So, is whole project in refactoring state now? I noticed tons of strange code and incomprehensible architecture decisions. :)

**Comment by benjaoming** [comment]
Is this a complaint or an issue? You're welcome to submit a PR expressing your wishes of improvement. Otherwise please open up one issue per problem and try to explain the problems instead of just throwing links to lines without actually stating the nature of the problems.

**Comment by DmitryFillo** [comment]
I explained the problem. I'm using mediabackup and run it with --clean key, but it doesn't work. It's an issue.

**Comment by benjaoming** [comment]
Sorry @DmitryFillo but you need to explain things more carefully otherwise it's impossible to understand what you mean.

**Comment by DmitryFillo** [comment]
1. I want to backup my media files.
2. My settings.py contains DBBACKUP_CLEANUP_KEEP_MEDIA = 2
3. I run ./manage.py mediabackup -c more than 10 times.
4. All backup files, that must be removed by DBBACKUP_CLEANUP_KEEP_MEDIA setting, not removed. All files are present in the filesystem.

I conclude that the DBBACKUP_CLEANUP_KEEP_MEDIA does not work. I figured out that get_backup_file_list() has wrong regexp or need to use os.path.basename(). I wrote about it in the first comment.

Why I did not pull request? Because I'm think that many part of the code must be rewrited and I have not enought time to figure out just now, but I want to report the bug.
